bin/swapSsmId.py

```diff
@@ -5,20 +5,31 @@
 
 if __name__ == '__main__':
 
-    parser = argparse.ArgumentParser(description="Swap ssmId's that are strings into integers.")
+    parser = argparse.ArgumentParser(description="Swap ssmId's that are strings into integers.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
     parser.add_argument("inputFile", type=str, help="Input file containing the DiaSources.")
     parser.add_argument("-t", "--trackId", type=str, default='ssmId_to_int.dat',
                         help="File tracking the map from the ssmId strings to integers.")
     parser.add_argument("-o", "--outFile", type=str, default=None,
                         help="Output file name for diasources with integer ssmIds. "\
                         "Default value is the input file, with addition of '_newId'.")
+    parser.add_argument("-k", "--ssmidKey", type=str, default=None,
+                        help="Name of the input file column containing SSMIDs."\
+                        "If not specified, will use a list of default key names and look"\
+                        "for the correct key.")
     args = parser.parse_args()
 
     diasources = pd.read_table(args.inputFile, delim_whitespace=True)
-    ssmKeys = ['objId', 'ssmId', 'ObjID', 'objID', 'ssmID']
-    for k in ssmKeys:
-        if k in diasources:
-            ssmKey = k
+
+    # If ssmidKey is given on command-line then use that.
+    # If it's not given, use list of defaults and find key.
+    if args.ssmidKey is not None:
+        ssmKey = args.ssmidKey
+    else:
+        ssmKeys = ['objId', 'ssmId', 'ObjID', 'objID', 'ssmID']
+        for k in args.ssmidKeys:
+            if k in diasources:
+                ssmKey = k
```

Comment:
I am confused a bit by this loop. It seems that `ssmKey` ends up being a value that is in both `diasources` and `ssmKeys`. My issues are:

- `ssmKey` can end up not being defined.
- if a command line value is given there is no check it is in `diasources`
- The loop should stop running once a value has been assigned to `ssmKey`.

In some sense `ssmKeys` could be defined as being an intersection of `diasources` and that list currently in `ssmKeys`. Then the code in line 30 is a simple test and no for loop required.